### ISSUE TITLE
Two build errors in win10 vs2017

### ISSUE THREAD

**Issue by Fish201809** [SFTtech/openage#1062]
error 1:
    3>c:\program files (x86)\microsoft visual `g:\workstation\current\openage\libopenage\log\logsink.cpp(66): error C2039: 'min': is not a member of 'std'`

The error was resolved when I tried to include `#include <algorithm>`

error 2:
  `  LINK : fatal error LNK1104: cannot open file 'pthread.lib'`

I try to run the command [`vcpkg install pthread`], But mistakes remain, I'm not sure how to correct the cmake file or something else

**Comment by TheJJ** [comment]
Hm. I fixed the first error in #1067. The second is strange, because I think on Windows openage should not use pthreads (those are for posix, hence the p). @tusharpm any hints how this should be handled? Currently `libopenage` does always link against `pthread` (via `libopenage/CMakeLists.txt`).

**Comment by tusharpm** [comment]
I think we should use something like
```cmake
find_package(Threads)
target_link_libraries(libopenage PUBLIC Threads::Threads [...])
```
instead of `pthread` in the mentioned file.
